widlparser/constructs.py

```diff
@@ -1869,11 +1869,11 @@ def attribute(self) -> str:
 
 	@property
 	def name(self) -> Optional[str]:
-		return cast(ChildProduction, self.parent).name if ('constructor' == self.idl_type) else self.attribute
+		return cast(ComplexProduction, self.parent).name if ('constructor' == self.idl_type) else self.attribute
```

Comment:
```suggestion
		return self.parent.name if ('constructor' == self.idl_type) else self.attribute
```
No need for the cast if parent can't be None

Reply:
Yup, just caught those while fixing up the symbol_table bit.